### PR TITLE
MAT-1932

### PR DESCRIPTION
**Pull request by willpowelson**
To convert Maccor protocol files to Biologic Protocol files we will need to be able to control what I'm calling "procedural steps" to sequences as opposed to the other type of Maccor step - "control steps".

Control steps are any of the following:
- Advancing Cycle Number
- Looping
- Do statements (beginning a cycle)

All other steps are procedural steps.

Procedural steps have a "control" portion to them, in that when a `<ReportEntry>` limit is reached they will go to some other arbitrary step. Since the mapping of steps to sequences is not 1-1 we'll need a more global context about the correspondence between step number from the input and sequence number in the output.

This behavior will be handled separately in a PR that's pending for the actual conversion.

### CODE REVIEW

File: beep/protocol/maccor_to_biologic_mb.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    Collection of methods to convert maccor protocol files to biologic modulo bat protocol files
+    Differences in underlying hardware and software makes this impossible in some cases
+    or forces workarounds that may fail. Certain work-arounds may be attempted and are
+    documented under the conversion function.
+    """
+
+    def __init__(self):
+        self.blank_seq = {
+            # sequence number
+            "Ns": "0",
```
Comment (patrickherring-TRI) [inline beep/protocol/maccor_to_biologic_mb.py:19]: We might want to move this information into a .yaml file. See /beep/beep/protocol/protocol_schemas/arbin_schedule_schema.yaml

Comment (willpowelson) [reply beep/protocol/maccor_to_biologic_mb.py:19]: Ah, sure.

As a personal preference I prefer to keep as much context in a single file as possible to reduce cognitive load from switching between files but we should be consistant, will move this to a file named `biologic_modulo_bat_schema.yaml`

**Comment by willpowelson** [comment]
@patrickherring-TRI 
Could I have some guidance on how test coverage works? Will I need test cases for every single error path to pass that check?

**Comment by patrickherring-TRI** [comment]
@willpowelson All of the checks should pass, with the possible exception of coverage. In terms of test cases, I think the most important thing to capture in the tests are the intended functionality, not necessarily focusing on every method having a test (although that can be nice). If there are errors that you can foresee, having some kind of assert might help to catch problems in the future, but you don't need an exhaustive list of errors.

**Comment by patrickherring-TRI** [comment]
@willpowelson Regarding the test coverage, I think the higher level philosophy is test driven development, writing the test as part of writing the code. But I think that's pretty similar to your normal workflow :)

File: beep/protocol/maccor_to_biologic_mb.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    """
+
+    def __init__(self):
+        BIOLOGIC_SCHEMA = loadfn(
+            os.path.join(PROTOCOL_SCHEMA_DIR, "biologic_mb_schema.yaml")
+        )
+        schema = OrderedDict(BIOLOGIC_SCHEMA)
+        self.blank_seq = schema["blank_seq"]
+
+    def __proc_step_to_seq(self, test_step, seq_num):
```
Comment (patrickherring-TRI) [inline beep/protocol/maccor_to_biologic_mb.py:25]: I don't think the double leading underscore is necessary. The convention I have seen is just one underscore

Comment (willpowelson) [reply beep/protocol/maccor_to_biologic_mb.py:25]: Ah, I didn't realize this! Can do!

**Pull request merged**